src/main/java/hudson/plugins/jira/Updater.java

```diff
@@ -321,6 +281,79 @@ private String getScmComments(boolean wikiStyle,
         return comment.toString();
     }
 
+    protected String createScmChangeEntryDescription(Run<?, ?> run, Entry change, boolean wikiStyle,
```

Comment:
As you are refactoring already, could you try to split this big pile of code into two methods: wiki-style and non-wiki style? I think that would be a lot more readable and make introducing format changes easier.

Another way of doing that would be using some kind of templating to let the user define the format (still, the method split applies).